src/blocks/icon/edit.js

```diff
@@ -27,18 +27,19 @@ const MAX_ICON_SIZE = 400;
 
 export { MIN_ICON_SIZE, MAX_ICON_SIZE };
 
-/**
- * Block edit function
- */
 class Edit extends Component {
-	componentDidMount() {
-		// Randomized the default icon when first added.
-		const defaultIcons = [ 'aperture', 'gesture', 'scatter_plot', 'waves', 'blocks', 'coblocks', 'drafts', 'device_hub', 'marker' ];
-		const rand = defaultIcons[ Math.floor( Math.random() * defaultIcons.length ) ];
+	isStylePreview() {
+		return ( $( `#block-${ this.props.clientId }` ).parent().parent().hasClass( 'block-editor-block-preview__content' ) );
+	}
 
-		// CHeck if the icon is the default.
-		if ( this.props.attributes.iconRand === true ) {
-			this.props.setAttributes( { icon: rand, iconRand: false } );
+	componentDidMount() {
+		if ( this.isStylePreview() ) {
+			this.props.setAttributes( { icon: 'aperture' } );
```

Comment:
This conditional shouldn't be needed and we should avoid changing the content for the style preview. The point of the style preview is to see your content in the different styles.